runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -5783,19 +5794,31 @@ class INTERPRETER_CLASS
 		case J9VM_CONTINUATION_RETURN_FROM_MONITOR_ENTER:
 			break;
 		case J9VM_CONTINUATION_RETURN_FROM_OBJECT_WAIT: {
-			j9object_t waitObject = *(j9object_t *)(_sp + 3);
-			rc = tryEnterBlockingMonitor(REGISTER_ARGS, waitObject, J9VM_CONTINUATION_RETURN_FROM_OBJECT_WAIT);
-			if ((NULL != _currentThread->currentContinuation) && (EXECUTE_BYTECODE == rc)) {
-				waitObject = *(j9object_t *)(_sp + 3);
-				omrthread_monitor_t monitor = getMonitorForWait(_currentThread, waitObject);
-				monitor->count = _currentThread->currentContinuation->waitingMonitorEnterCount;
+			rc = tryEnterBlockingMonitor(REGISTER_ARGS, syncObject, J9VM_CONTINUATION_RETURN_FROM_OBJECT_WAIT);
+			J9VMContinuation *continuation = _currentThread->currentContinuation;
+			if ((NULL != continuation) && (EXECUTE_BYTECODE == rc)) {
+				syncObject = *(j9object_t *)(_sp + 3);
+				omrthread_monitor_t monitor = getMonitorForWait(_currentThread, syncObject);
+				monitor->count = continuation->waitingMonitorEnterCount;
 				_currentThread->ownedMonitorCount += monitor->count - 1;
-				_currentThread->currentContinuation->waitingMonitorEnterCount = 0;
+				continuation->waitingMonitorEnterCount = 0;
+
+				j9object_t threadObject = _currentThread->threadObject;
+				bool interrupted = J9VMJAVALANGTHREAD_DEADINTERRUPT(_currentThread, threadObject);
+				bool notified = J9VMJAVALANGVIRTUALTHREAD_NOTIFIED(_currentThread, threadObject);
+
+				if (J9_EVENT_IS_HOOKED(_vm->hookInterface, J9HOOK_VM_MONITOR_WAITED)) {
+					IDATA rc = (interrupted ? J9THREAD_INTERRUPTED : (notified ? 0 : J9THREAD_TIMED_OUT));
+					I_64 millis = J9VMJAVALANGVIRTUALTHREAD_TIMEOUT(_currentThread, threadObject);
```

Comment:
Again, the `timeout` field is considered to be nanoseconds.

Reply:
see above comment.